### ISSUE TITLE
Create Chat Page

### ISSUE THREAD

**Issue by Jessika63** [Jessika63/PharmaXcess#103]
Description: Develop the Chat page, enabling users to create support tickets for discussions with healthcare professionals. This includes creating and managing discussions, viewing open conversations, and closing discussions once complete.

Steps:

1.     Step 1: Add Ticket Creation Feature for Starting a Discussion

- [ ]         Allow users to create a ticket to initiate a new discussion.

- [ ]         Include fields for the subject or topic of the discussion to help in directing it to the appropriate support.

- [ ]         Provide clear prompts to guide the user in creating a ticket.

2.     Step 2: Send Ticket to Support

- [ ]         Implement functionality to securely send the ticket to the support system.

- [ ]         Ensure the ticket information is correctly recorded and securely transmitted.

- [ ]         Display a confirmation message to the user upon successful submission of the ticket.

3.     Step 3: Assign Ticket to a Competent Healthcare Professional

- [ ]         Set up a system to assign the ticket to a healthcare professional with relevant expertise based on the topic of discussion.

- [ ]         Ensure that each ticket is matched with an appropriate specialist for accurate and helpful responses.

- [ ]         Implement a notification system to inform the assigned professional of the new ticket.

4.     Step 4: Create the Discussion

- [ ]         Once a ticket is assigned, create a discussion thread that allows for back-and-forth messaging between the user and the healthcare professional.

- [ ]         Display messages in a clear chat interface to facilitate communication.

- [ ]         Include functionality for real-time message updates to create a responsive chat experience.

5.     Step 5: Close the Discussion When Finished

- [ ]         Add a feature that allows either the user or the healthcare professional to close the discussion once it is complete.

- [ ]         Confirm with the user before finalizing the closure.

- [ ]         Notify both parties that the discussion is now closed.

6.     Step 6: Display All Open or Ongoing Discussions on the Chat Page

- [ ]         Show a list of all active discussions on the Chat page for easy access.

- [ ]         Indicate the status of each discussion (e.g., "In Progress," "Waiting for Response").

- [ ]         Allow users to quickly enter any ongoing discussion from the list.

7.     Step 7: Remove Closed Discussions from the Chat Page

- [ ]         Once a discussion is closed, remove it from the list of active discussions on the Chat page.

- [ ]         Provide a separate archive or history view if users need access to past closed discussions.

- [ ]         Test the removal process to ensure closed discussions do not reappear in the active list.
